### ISSUE TITLE
Consolidate `DESCRIPTION.rst` and `README.md`

### ISSUE THREAD

**Issue by leofang** [NVIDIA/cuda-python#300]
> Indeed we're duplicating some of the information. It might take a few iterations to consolidate the two files and have something presentable on both PyPI and GitHub. Let's address this in the next release.

_Originally posted by @leofang in https://github.com/NVIDIA/cuda-python/issues/296#issuecomment-2542240128_

**Comment by vzhurba01** [comment]
I'd like to extend the scope of this issue to help consolidate/organize all the `README.md` and `DESCRIPTION.rst` between `cuda-python`, `cuda.bindings` and `cuda.core`. When resolving issue #297 with [PR 397](https://github.com/NVIDIA/cuda-python/pull/397) I found that `cuda.bindings` also has awkwardness in creating a consolidated file that is presentable on both PyPI and GitHub.

For PyPI, it seems like we'd want:
- To direct users to documentation
- Runtime dependencies
- Installing? (Perhaps a link to installing page in documentation is better)

For GitHub:
- Same as PyPI
- Development info
- Running tests, samples and benchmarks

There's some overlap in dev info across subpackages that may belong better under `cuda-python`:
- Code Linting
- Code Signing
- Supported Python Versions

In conclusion, info that applies to all subpackages should be under `cuda-python` while development info specific to a subpackage should be kept out of PyPI.

**Comment by leofang** [comment]
Sounds like a plan! You have earned the title "trusted assignee," Vlad 👍
